src/uncertainties.jl

```diff
@@ -0,0 +1,36 @@
+## New Type: DataPoint
+# Struct for carrying statistic and systematic uncertainty of a DataPoint
+struct DataPoint{T<:Number} <: Number
+    val::T
+    stat_err::Tuple{T,T}
+    sys_err::Tuple{T,T}
+end
+
+# Helper functions
+_to_tuple(x::T where T<:Number) = (x,x)
+_to_tuple(x::Tuple) = promote(x...)
+
+# Constructors
+DataPoint(x; stat=zero(x), sys=zero(x)) = DataPoint(x, _to_tuple(stat), _to_tuple(sys))   # with keywords
+DataPoint(x, stat, sys = zero(x)) = DataPoint(x, stat = stat, sys = sys) # ordered without keywords
+const ±(val, err) = DataPoint(val, err)
+const ±(val, err::NamedTuple) = DataPoint(val, stat = err.stat, sys = err.sys)
```

Comment:
I wonder if we can create something that works like
```julia
datapoint = val ± sys ± stat
```
but maybe that's for another PR.

Reply:
Yes this is possible because this would be parsed as 
```julia
datapoint = val ± stat ± sys  # (val ± stat) ± sys
```
which is calling first `±(val, stat) = DataPoint(val,stat)` which returns a `DataPoint`. So by adding a method `±(::DataPoint,sys)` this can be done.

Reply:
I was thinking about how we could make it that `val ± stat ± sys` gets parsed as `±(val,stat,sys)` which we can then dispatch into the constructor.

Reply:
Operators are always binary so there is no (easy) way to do this.
The reason why such functions exist for e.g. `+(x,y,z...)` is because this doesn't change the type and calls itself iteratively like `+(+(+(x,y),z),...)`.

The function I had in mind (which also works) is:
```julia
const ±(D::DataPoint, sys) = DataPoint(D.val, D.stat_err, sys)
```
however here the order is really important.